server/app/export.go

```diff
@@ -0,0 +1,208 @@
+package app
+
+import (
+	"archive/zip"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/mattermost/focalboard/server/model"
+	"github.com/mattermost/focalboard/server/services/store"
+	"github.com/wiggin77/merror"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+var (
+	newline = []byte{'\n'}
+)
+
+func (a *App) ExportArchive(w io.Writer, opt model.ExportArchiveOptions) (errs error) {
+	container := store.Container{
+		WorkspaceID: opt.WorkspaceID,
+	}
+	boards, err := a.getBoardsForArchive(container, opt.BoardIDs)
+	if err != nil {
+		return err
+	}
+
+	merr := merror.New()
+	defer func() {
+		errs = merr.ErrorOrNil()
+	}()
+
+	// wrap the writer in a zip.
+	zw := zip.NewWriter(w)
+	defer func() {
+		merr.Append(zw.Close())
+	}()
+
+	if err := a.writeArchiveVersion(zw); err != nil {
+		merr.Append(err)
+		return
+	}
+
+	for _, board := range boards {
+		if err := a.writeArchiveBoard(zw, board, opt); err != nil {
+			merr.Append(fmt.Errorf("cannot export board %s: %w", board.ID, err))
+			return
+		}
+	}
+	return nil
+}
+
+// writeArchiveHeader writes a version file to the zip.
+func (a *App) writeArchiveVersion(zw *zip.Writer) error {
+	archiveHeader := model.ArchiveHeader{
+		Version: archiveVersion,
+		Date:    model.GetMillis(),
+	}
+	b, _ := json.Marshal(&archiveHeader)
+
+	w, err := zw.Create("version.json")
+	if err != nil {
+		return fmt.Errorf("cannot write archive header: %w", err)
+	}
+
+	if _, err := w.Write(b); err != nil {
+		return fmt.Errorf("cannot write archive header: %w", err)
+	}
+	return nil
+}
+
+// writeArchiveBoard writes a single board to the archive in a zip directory.
+func (a *App) writeArchiveBoard(zw *zip.Writer, board model.Block, opt model.ExportArchiveOptions) error {
+	// create a directory per board
+	w, err := zw.Create(board.ID + "/board.jsonl")
+	if err != nil {
+		return err
+	}
+
+	// write the board block first
+	if err = a.writeArchiveBlockLine(w, board); err != nil {
+		return err
+	}
+
+	var files []string
+	container := store.Container{
+		WorkspaceID: opt.WorkspaceID,
+	}
+
+	// write the content blocks
+	// TODO: paginate this
+	contentBlocks, err := a.GetBlocksWithRootID(container, board.ID)
```

Comment:
nit: This is also missleading. there are blocks in board that are content (like text, comment, image), but also you have `views`

Reply:
Fixed via https://github.com/mattermost/focalboard/pull/2201/commits/4635eb5b6b79391af0ab7f7ecc008d3a7d7704a8